### ISSUE TITLE
No way to send raw data in post requests in vegeta

### ISSUE THREAD

**Issue by fengnex** [tsenart/vegeta#530]
#### Version and Runtime

```
Paste the output of `vegeta -version` here.
If you are not running the latest version of Vegeta, please try upgrading because your issue may have already been fixed.
```
12.8.0

#### Expected Behaviour
Sending raw data to the server in post requests through vegeta, and the server can receive request data, whenever curl can work out with the same effect.

When we execute the following curl command, the server can successfully receive the raw data that is specified with option -d. The raw data is a plain string, instead of normal key=value string.
```
curl -H 'eq:Y6IxBROYbOAA5RxSJSCbUw==' -X POST https://testURL -d '6+7JxfoPsvEsjADT8WWYFx1PJRmeVle1UXZJGBniea5J3DuB8OV67JAH1Uty296rTy0ReTBU4Sf1WcAPmN7WPGadp76DWypQKoKOxA4TYUtfJnrfSWndtxvm+VNU4h7u84Ao2t4kd3eFtccIu95C784bZx4F9lSICS6R0eVVHNj5zSn/PkwfiMS7rczSPL1Q0Ixn2rCgmG6zbrqvzw+PrOzSb/spclDG9BM2Xgjpe/ydXpQXUV50Ncg6CYLOKEU7qDSe4JQiKycHGnKFLljv6uzhJWvlIXsvyGF7B6BT6jM0BHF+z45W1j0rNDIH9Zxx6jyEel7FDHBA44K4KnF2PsyecRgDuY1UV579pAZjCji0UFb3Lsu2jJiLT6gcjG3j5pJgWY85H2C+fLjfU48BY9X4iWtJpeoISHwbqJ/4kzE1tdVUGN4fwWLYx5afX4bcbZbzMelAnXcK+huaUA/GLOqR2neNIfifBrg1MLcUW+BNorAJwfRJIf0/Lvc0t+q/'
```

What we expect to see in vegeta is that the sever can also receive the raw data string when we execute the following vegeta commands:
```
echo "6+7JxfoPsvEsjADT8WWYFx1PJRmeVle1UXZJGBniea5J3DuB8OV67JAH1Uty296rTy0ReTBU4Sf1WcAPmN7WPGadp76DWypQKoKOxA4TYUtfJnrfSWndtxvm+VNU4h7u84Ao2t4kd3eFtccIu95C784bZx4F9lSICS6R0eVVHNj5zSn/PkwfiMS7rczSPL1Q0Ixn2rCgmG6zbrqvzw+PrOzSb/spclDG9BM2Xgjpe/ydXpQXUV50Ncg6CYLOKEU7qDSe4JQiKycHGnKFLljv6uzhJWvlIXsvyGF7B6BT6jM0BHF+z45W1j0rNDIH9Zxx6jyEel7FDHBA44K4KnF2PsyecRgDuY1UV579pAZjCji0UFb3Lsu2jJiLT6gcjG3j5pJgWY85H2C+fLjfU48BY9X4iWtJpeoISHwbqJ/4kzE1tdVUGN4fwWLYx5afX4bcbZbzMelAnXcK+huaUA/GLOqR2neNIfifBrg1MLcUW+BNorAJwfRJIf0/Lvc0t+q/" > body.txt

echo "POST http://testURL" | vegeta attack -rate 50/s -duration 1m -body body.txt -header "eq: Y6IxBROYbOAA5RxSJSCbUw==" -header "Content-Type: text/plain" > results.bin

vegeta report results.bin
```
#### Actual Behaviour
No data has been received in the server side system.

#### Steps to Reproduce

1. echo "6+7JxfoPsvEsjADT8WWYFx1" > body.txt
2. echo "POST http://testURL" | vegeta attack -rate 1/s -duration 10s -body body.txt -header "eq: Y6IxBROYbOAA5RxSJSCbUw==" -header "Content-Type: text/plain" > results.bin &
3. Check the server system and its logs to verify whether it has received the requests data: "6+7JxfoPsvEsjADT8WWYFx1"

**Comment by tsenart** [comment]
@fengnex: You can easily check whether the problem is with Vegeta or your server by running that attack against this test server in the Vegeta repo (it requires you have Go installed, let me know if you want a static binary instead):

```
git clone https://github.com/tsenart/vegeta
cd vegeta
go run ./internal/cmd/echosrv/main.go :6060 -dump
```

Then in another tab or terminal window, run the attack against `localhost:6060` and see if your request body is printed to stdout. If it is, then you have a problem with your server, because vegeta is actually sending the request body. Otherwise, I'll look into it.

**Comment by fengnex** [comment]
@tsenart Thank you! I'm afraid that you may want to look into it, and please.

This is because when attacking internal/cmd/echosrv/main.go with the following command:
echo "POST http://localhost:6060" | vegeta attack -rate 1/s -duration 5s -body body.txt > results.bin

the output we got looks like that displayed in the following image, and there are actually no dumped requests data has been printed out :
![NIUJ{Q{FSDBA_I%XB}VL DE](https://user-images.githubusercontent.com/26297547/87108596-43b3e480-c295-11ea-81ab-7329ec949ad5.png)

**Comment by tsenart** [comment]
Sorry, the instructions I sent you to run the test server were wrong, we need to pass the `-dump` flag before the listen address. I don't know how you got that output for curl.

Run this instead `go run ./internal/cmd/echosrv/main.go -dump :6060` and try again.

**Comment by fengnex** [comment]
@tsenart You are right. The output of curl and vegeta's attacking are highly similar to each other. We need to have further verification.

**Comment by fengnex** [comment]
Vegeta is on in the scenario we have discussed and it has no relationship with the Accept-Encoding and other http header options.
In this case, there is a character \x0A (new line) automatically added at the end of the parameter in the context of vegeta, which causes the server cannot process the string successfully.
